bool[] data_1d